specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/stable/2023-09-04/redhatopenshift.json

```diff
@@ -0,0 +1,2566 @@
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The List of available versions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OpenShiftVersion"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "description": "Next Link to next operation.",
+          "type": "string"
+        }
+      }
+    },
+    "OpenShiftVersionProperties": {
+      "description": "OpenShiftVersionProperties represents the properties of an OpenShiftVersion.",
+      "type": "object",
+      "properties": {
+        "version": {
+          "description": "Version represents the version to create the cluster at.",
+          "type": "string"
+        }
+      }
+    },
+    "Operation": {
+      "description": "Operation represents an RP operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Operation name: {provider}/{resource}/{operation}.",
+          "type": "string"
+        },
+        "display": {
+          "$ref": "#/definitions/Display",
+          "description": "The object that describes the operation."
+        },
+        "origin": {
+          "description": "Sources of requests to this operation.  Comma separated list with valid values user or system, e.g. \"user,system\".",
+          "type": "string"
+        }
+      }
+    },
+    "OperationList": {
+      "description": "OperationList represents an RP operation list.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of operations supported by the resource provider.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of operations.",
+          "type": "string"
+        }
+      }
+    },
+    "OutboundType": {
+      "description": "The outbound routing strategy used to provide your cluster egress to the internet.",
+      "enum": [
+        "Loadbalancer",
+        "UserDefinedRouting"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "OutboundType",
+        "modelAsString": true
+      }
+    },
+    "PreconfiguredNSG": {
+      "description": "PreconfiguredNSG represents whether customers want to use their own NSG attached to the subnets",
+      "enum": [
+        "Disabled",
+        "Enabled"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "PreconfiguredNSG",
+        "modelAsString": true
+      }
+    },
+    "ProvisioningState": {
+      "description": "ProvisioningState represents a provisioning state.",
+      "enum": [
+        "AdminUpdating",
+        "Creating",
+        "Deleting",
+        "Failed",
+        "Succeeded",
+        "Updating"
```

Comment:
ProvisioningState is missing a required value of "Canceled".

Reply:
This is out of scope for now and is considered for future.

Reply:
Why not now?

Reply:
@gouthamMN Hey Goutham. We upstreamed this to ARO-RP. We can include it now to pass review. Please use https://github.com/Azure/ARO-RP/pull/3117 as a guide.

Reply:
updated